modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc

```diff
@@ -0,0 +1,137 @@
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      externalLabels:
+        <key>: <value> <1>
+----
++
+<1> Substitute `<key>: <value>` with a map of key-value pairs where `<key>` is a unique name for the new label and `<value>` is its value.
++
+[WARNING]
+====
+Do not use `prometheus` or `prometheus_replica` as key names, because they are reserved and will be overwritten.
+====
++
+For example, to add metadata about the region and environment to all time series and alerts, use:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      externalLabels:
+        region: eu
+        environment: prod
+----
+
+** *To attach custom labels to all time series and alerts leaving the Prometheus instance that monitors user-defined projects*:
+.. Edit the `user-workload-monitoring-config` `ConfigMap` object in the `openshift-user-workload-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
+----
+
+.. Define a map of labels you want to add for every metric under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    prometheus:
+      externalLabels:
+        <key>: <value> <1>
+----
++
+<1> Substitute `<key>: <value>` with a map of key-value pairs where `<key>` is a unique name for the new label and `<value>` is its value.
++
+[WARNING]
+====
+Do not use `prometheus` or `prometheus_replica` as key names, because they are reserved and will be overwritten.
+====
++
+[NOTE]
+====
+In the `openshift-user-workload-monitoring` project, Prometheus handles metrics and Thanos Ruler handles alerting and recording rules. Setting `externalLabels` for `prometheus` in the `user-workload-monitoring-config` `ConfigMap` object will only configure external labels for metrics and not for any rules.
+====
++
+For example, to add metadata about the region and environment to all time series and alerts related to user-defined projects, use:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    prometheus:
+      externalLabels:
+        region: eu
+        environment: prod
+----
+
+. Save the file to apply the changes. The new configuration is applied automatically.
++
+[NOTE]
+====
+Configurations applied to the `user-workload-monitoring-config` `ConfigMap` object are not activated unless a cluster administrator has enabled monitoring for user-defined projects.
```

Comment:
This is already enabled for OSD/ROSA, so this note is not needed

Reply:
Removed.